Running this locally I get:

```
[10:07] ~/git/silex-trailing-slash-handler ● make 'composer-update --prefer-stable --prefer-dist --prefer-lowest'
...
[10:13] ~/git/silex-trailing-slash-handler ● make test
docker run --rm -it -v $(pwd):/opt/graze/silex-trailing-slash-handler graze/silex-trailing-slash-handler \
	composer test --ansi
Do not run Composer as root/super user! See https://getcomposer.org/root for details
> vendor/bin/phpunit --colors=always tests/
PHPUnit 5.1.1 by Sebastian Bergmann and contributors.

...............................
Fatal error: Call to undefined method Symfony\Component\Debug\ExceptionHandler::getHtml() in /opt/graze/silex-trailing-slash-handler/vendor/silex/silex/src/Silex/ExceptionHandler.php on line 44
Script vendor/bin/phpunit --colors=always tests/ handling the test event returned with error code 255
make: *** [test-unit] Error 255
```

I guess something else needs updating to a minimum version

--- a/composer.json
+++ b/composer.json
@@ -29,7 +29,7 @@
     },
 
     "require": {
-        "silex/silex": "^1.2"
+        "silex/silex": "^2"
     },
 
     "require-dev": {

--- a/src/TrailingSlashControllerProvider.php
+++ b/src/TrailingSlashControllerProvider.php
@@ -2,9 +2,10 @@
 
 namespace Graze\Silex\ControllerProvider;
 
+use Pimple\Container;
+use Pimple\ServiceProviderInterface;
 use Silex\Application;
-use Silex\ControllerProviderInterface;
-use Silex\ServiceProviderInterface;
+use Silex\Api\ControllerProviderInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpKernel\HttpKernelInterface;
 use Symfony\Component\Routing\Matcher\UrlMatcher;
@@ -75,29 +76,20 @@ final class TrailingSlashControllerProvider implements ControllerProviderInterfa
     }
 
     /**
-     * @param Application $app
+     * @param Container $app
      */
-    public function register(Application $app)
+    public function register(Container $app)
     {
         /**
          * We override the default RedirectableUrlMatcher so that Silex doesn't
          * respond with 301 to GET requests missing a trailing slash.
          */
-        $app['url_matcher'] = $app->share(
-            function () use ($app) {
-                if ($app['logger']) {
-                    $app['logger']->debug(sprintf('Overriding the default Silex url matcher to %s.', UrlMatcher::class));
-                }
-
-                return new UrlMatcher($app['routes'], $app['request_context']);
+        $app['url_matcher'] = function (Container $app) {
+            if ($app['logger']) {
+                $app['logger']->debug(sprintf('Overriding the default Silex url matcher to %s.', UrlMatcher::class));
             }
-        );
-    }
 
-    /**
-     * @param Application $app
-     */
-    public function boot(Application $app)
-    {
+            return new UrlMatcher($app['routes'], $app['request_context']);
+        };
     }
 }

--- a/src/TrailingSlashControllerProvider.php
+++ b/src/TrailingSlashControllerProvider.php
@@ -40,20 +40,21 @@ final class TrailingSlashControllerProvider implements ControllerProviderInterfa
                 $app['logger']->debug(sprintf('Appending a trailing slash for the request to `/%s`.', $resource));
             }
 
+            $currentRequest = $app['request_stack']->getCurrentRequest();
             $request = Request::create(
                 '/' . $resource . '/',
-                $app['request']->getMethod(),
+                $currentRequest->getMethod(),
                 [],
-                $app['request']->cookies->all(),
-                $app['request']->files->all(),
-                $app['request']->server->all(),
-                $app['request']->getContent()
+                $currentRequest->cookies->all(),
+                $currentRequest->files->all(),
+                $currentRequest->server->all(),
+                $currentRequest->getContent()
             );
             $request = $request->duplicate(
-                $app['request']->query->all(),
-                $app['request']->request->all()
+                $currentRequest->query->all(),
+                $currentRequest->request->all()
             );
-            $request->headers->replace($app['request']->headers->all());
+            $request->headers->replace($currentRequest->headers->all());
 
             // Make an internal sub-request based off the request that would have 404'd.
             // http://silex.sensiolabs.org/doc/usage.html#forwards

--- a/tests/TrailingSlashControllerProviderTest.php
+++ b/tests/TrailingSlashControllerProviderTest.php
@@ -4,10 +4,10 @@ namespace Graze\Silex\Tests\ControllerProvider;
 
 use Graze\Silex\ControllerProvider\TrailingSlashControllerProvider;
 use Mockery;
+use Pimple\ServiceProviderInterface;
 use Psr\Log\LoggerInterface;
 use Silex\Application;
-use Silex\ControllerProviderInterface;
-use Silex\ServiceProviderInterface;
+use Silex\Api\ControllerProviderInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\Routing\Matcher\UrlMatcher;
 

